stm32-modules/heater-shaker/firmware/host_comms_task/freertos_comms_task.cpp

```diff
@@ -116,14 +123,24 @@ void run(void *param) {  // NOLINT(misc-unused-parameters)
                                &local_task->cdc_class_fops);
     USBD_SetClassConfig(&local_task->usb_handle, 0);
     USBD_Start(&local_task->usb_handle);
+    UART_Init(&local_task->uart_handle);
     local_task->committed_rx_buf_ptr = local_task->rx_buf.committed()->data();
+    local_task->committed_uart_rx_buf_ptr = local_task->uart_rx_buf.committed()->data();
+    HAL_UART_Receive_IT(
+        &_local_task.uart_handle,
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+        reinterpret_cast<uint8_t *>(
+            _local_task.uart_rx_buf.committed()->data()),
+        (uint16_t)(
+            _local_task.uart_rx_buf.committed()->size()));
     while (true) {
         char *tx_end =
             top_task->run_once(local_task->tx_buf.accessible()->begin(),
                                local_task->tx_buf.accessible()->end());
         if (!top_task->may_connect()) {
             USBD_Stop(&_local_task.usb_handle);
-        } else if (tx_end != local_task->tx_buf.accessible()->data()) {
+            UART_DeInit(&_local_task.uart_handle);
+        } else if ((tx_end != local_task->tx_buf.accessible()->data()) && (UartReady == SET)) {
```

Comment:
this makes me a little nervous; the UART is entirely a secondary capacity, and this will block USB output on the uart being ready. We should either use a doublebuf on the tx side for the uart or go full DMA or something so we can remove the `(UartReady == SET)` condition.